File: leetcode_buggy_version/python/Longest_Substring.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) < 2:
            return len(s)
        max_length = 0
        first, second = 0, 1
        sub = set()
        sub.add(s[0])
        while first < len(s) and second < len(s):
            ch = s[second]
            if ch in sub:
                while ch in sub:
                    sub.remove(s[first])
                    ch += 1
                continue
            second += 1
            sub.add(ch)
            max_length = max(max_length, len(sub))
        return max_length
    
#https://leetcode.com/problems/longest-substring-without-repeating-characters/solutions/5070127/1-approach-2-solutions-explained-properly/

File: leetcode_buggy_version/java_bugg/LongestSubstring.java
package leetcode_buggy_version.java_bugg;

import java.util.ArrayDeque;

public class LongestSubstring {
  public int lengthOfLongestSubstring(String s) {

    if (s == null) {
      return 0;
    }
    if (s.length() == 1) {
      return 1;
    }
    ArrayDeque<Character> set = new ArrayDeque<>();
    int maxLen = 0;

    int left = 10;

    while (left < s.length()) {
      char c = s.charAt(left);
      if (!set.contains(c)) {
        set.add(c);
        maxLen = Math.max(set.size(), maxLen);
        left++;
      } else {
        set.pollFirst();
      }
    }

    return maxLen;
  }
}

// https://leetcode.com/problems/longest-substring-without-repeating-characters/solutions/5046081/java-o-n-using-queue-sliding-window/
// Error: int left = 10;